lib/Server.js

```diff
@@ -552,6 +552,58 @@ class Server {
           searchParams.set("password", webSocketURL.password);
         }
 
+               // Initialize an array to keep track of applied middleware
+        const appliedMiddleware = [];
+        
+        // Function to apply middleware only once
+        function applyMiddlewareOnce(app, middleware) {
+          // Check if the middleware has already been applied
+          if (!appliedMiddleware.includes(middleware)) {
+            // Apply the middleware
+            app.use(middleware);
+            
+            // Add the middleware to the applied middleware array
+            appliedMiddleware.push(middleware);
+          }
+        }
+        
+        // Test cases for applyMiddlewareOnce function
+        describe('applyMiddlewareOnce', () => {
+          let app;
+        
+          beforeEach(() => {
+            // Mock Express app
+            app = {
+              use: jest.fn()
+            };
+          });
+        
+          afterEach(() => {
+            jest.clearAllMocks();
+          });
+        
+          it('should apply middleware only once', () => {
+            const middlewareFunction = jest.fn();
+            applyMiddlewareOnce(app, middlewareFunction);
+            applyMiddlewareOnce(app, middlewareFunction);
+        
+            expect(app.use).toHaveBeenCalledTimes(1);
+            expect(app.use).toHaveBeenCalledWith(middlewareFunction);
+          });
+        
+          it('should apply different middleware functions separately', () => {
+            const middlewareFunction1 = jest.fn();
+            const middlewareFunction2 = jest.fn();
+            applyMiddlewareOnce(app, middlewareFunction1);
+            applyMiddlewareOnce(app, middlewareFunction2);
+        
+            expect(app.use).toHaveBeenCalledTimes(2);
+            expect(app.use).toHaveBeenCalledWith(middlewareFunction1);
+            expect(app.use).toHaveBeenCalledWith(middlewareFunction2);
+          });
+        });
```

Comment:
We should not store it here

Reply:
So please suggest for proceedings

Reply:
We have the `test` directory

Reply:
can you suggest me file name , it can be big help